src/main/java/org/junitpioneer/jupiter/AbstractEntryBasedExtension.java

```diff
@@ -43,60 +45,52 @@
  * @param <S> The set annotation type.
  */
 abstract class AbstractEntryBasedExtension<K, V, C extends Annotation, S extends Annotation>
-		implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback {
-
-	@Override
-	public void beforeAll(ExtensionContext context) {
-		clearAndSetEntries(context);
-	}
+		implements BeforeEachCallback, AfterEachCallback {
 
 	@Override
 	public void beforeEach(ExtensionContext context) {
-		clearAndSetEntries(context);
+		List<ExtensionContext> contexts = PioneerUtils.findAllContexts(context);
```

Comment:
Honestly, just return with a `Stream` from findAllContexts, so you can write:
```java
PioneerUtils.findAllContexts(context)
    .sorted(Collections.reverseOrder()) // Could be static imported, so it's just reverseOrder()
    .forEach(this::clearAndSetEntries);
```

Reply:
[Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#reverseOrder--):

>Returns a comparator that imposes the reverse of the _natural ordering_ on a collection of objects that implement the `Comparable` interface.

We don't want the list to be sorted (and even if, `ExtensionContext` isn't comparable). Instead, we want to reverse the entire list. This isn't straightforward for streams (see ["Java 8 stream reverse order"](https://stackoverflow.com/q/24010109)), so I decided to return a list.

OK for you?

Reply:
My bad 😅